src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Publish.targets

```diff
@@ -770,6 +846,29 @@ Copyright (c) .NET Foundation. All rights reserved.
 
   </Target>
 
+  <!--
+    ============================================================
+                                        BundlePublishDirectory
+
+    Bundle the items in PreBundlePublishDir into one file in PublishDir
+    ============================================================
+    -->
+  <Import Project="Sdk.props" Sdk="Microsoft.NET.Build.Bundle" />
+
+  <Target Name="BundlePublishDirectory" 
+          Condition="'$(PublishSingleFile)' == 'true'"
+          Inputs="$(PreBundlePublishDir)\**\*.*"
```

Comment:
Per our discussion I don't think wildcard syntax works for includes.  You'll need to run a target before and evaulate this into an item.